app/models/region.rb

```diff
@@ -10,11 +12,11 @@ class Region < ApplicationRecord
 
   belongs_to :source, polymorphic: true, optional: true
 
-  before_discard do
-    self.path = nil
-  end
-
-  MAX_LABEL_LENGTH = 255
+  # To set a new path for a Region, assign the parent region via `reparent_to`, and the before_validation
+  # callback will assign the new path.
+  attr_accessor :reparent_to
```

Comment:
Can we continue using the `@parent` writer method instead of adding a `reparent_to`? imo `parent:` makes for a sweet API, given that we're adding this method just to make setting paths ergonomic.

Reply:
We discussed in a zoom, but just for posterity - the goal here was to keep the destructive api of 'reparent' - i.e. setting a parent and changing the path as very separate from the read only API of retrieving the parent.